src/challeng.cpp

```diff
@@ -565,34 +540,32 @@ int fighting_duel(CHAR_DATA *ch, CHAR_DATA *victim) {
     return true;
 }
 
-int in_duel(CHAR_DATA *ch) {
+int in_duel(const CHAR_DATA *ch) {
     if (!challenge_fighting)
         return false;
     if (ch->in_room->vnum != CHAL_ROOM)
         return false;
-    if ((ch != challenger) && (ch != challengee))
-        return false;
-    return true;
+    return ch == challenger || ch == challengee;
 }
 
 /* Checks to see that various things are correct.  Only input is
    toggle to say what to do on error.  0=nothing, 1=abort duel. */
 static int check_duel_status(int f) {
-    int flag = 0;
+    unsigned int flag = 0;
 
     if (challenger != nullptr) {
-        if (strncmp(challenger->name, challenger_name, NAME_SIZE))
-            flag = 1;
+        if (challenger->name == challenger_name)
```

Comment:
is the logic on these == right?  `if(strncmp` will only return 0 if it's equal?

Reply:
Wow! Great spot: I was 100% wrong here!!!

Reply:
Fixed! Am going to go over everything one more time then I'll merge!